controllers/chat/ChatController.js

```diff
@@ -462,23 +467,61 @@ module.exports = {
     };
 
     try {
-      if (!members.includes(req.userId)) members.push(req.userId);
       await client.connectUser({id: req.userId}, req.token);
+      let channel = null;
+      // check if member are both anonymous
+      if (!members.includes(req.userId)) members.push(req.userId);
+      let is_anon_to_anon = await is_all_anon_user(members);
+      if (is_anon_to_anon) {
+        // check if channel already exist
+        let new_channel = await handle_anon_to_anon_channel_owner(req.userId, members[0], context);
+        await handle_anon_to_anon_channel_member(new_channel, members[0], oldChannelId, context);
+        channel = client.channel('messaging', new_channel.channel_id, {
+          members
+        });
+      } else {
+        channel = client.channel('messaging', {
+          members
+        });
+      }
 
-      const channel = client.channel('messaging', {
-        members
-      });
       const findOrCreateChannel = await channel.create();
       findOrCreateChannel.members = await Promise.all(
         findOrCreateChannel.members.map(async (member) => {
-          if (member.role !== 'owner') return member;
+          if (member.role !== 'owner' && !member.user?.username?.includes('Anonymous'))
+            return member;
 
-          const anonInfo = await ChatAnonUserInfo.findOne({
+          let anonInfo = await ChatAnonUserInfo.findOne({
             where: {
               channel_id: findOrCreateChannel.channel.id,
               my_anon_user_id: member.user_id
             }
           });
+          if (!anonInfo && oldChannelId && member.user_id !== req.userId) {
+            let oldAnonInfo = await ChatAnonUserInfo.findOne({
+              where: {
+                channel_id: oldChannelId,
+                my_anon_user_id: {
+                  [Sequelize.Op.ne]: req.userId
+                }
+              }
+            });
+            if (oldAnonInfo) {
+              anonInfo = oldAnonInfo;
+              await ChatAnonUserInfoFunction.createChatAnonUserInfo(
+                ChatAnonUserInfo,
+                findOrCreateChannel.channel.id,
+                member.user_id,
+                req?.userId,
+                {
+                  anon_user_info_color_code: anonInfo.anon_user_info_color_code,
+                  anon_user_info_color_name: anonInfo.anon_user_info_color_name,
+                  anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
+                  anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name
+                }
+              );
+            }
+          }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for mapping `findOrCreateChannel.members` has been updated to handle anonymous users. If conditions are met, it searches for an existing `ChatAnonUserInfo` record or creates a new one based on an `oldChannelId`.
```diff
-      const findOrCreateChannel = await channel.create();
-      findOrCreateChannel.members = await Promise.all(
-        findOrCreateChannel.members.map(async (member) => {
-         if (member.role !== 'owner') return member;
- 
-         const anonInfo = await ChatAnonUserInfo.findOne({
-            where: {
-              channel_id: findOrCreateChannel.channel.id,
-              my_anon_user_id: member.user_id
-            }
-          });
- 
-          let anonUserInfo = {};
+      const findOrCreateChannel = await channel.create();
+      findOrCreateChannel.members = await Promise.all(
+        findOrCreateChannel.members.map(async (member) => {
+          if (member.role !== 'owner' && !member.user?.username?.includes('Anonymous'))
+            return member;
+  
+          let anonInfo = await ChatAnonUserInfo.findOne({
+            where: {
+              channel_id: findOrCreateChannel.channel.id,
+              my_anon_user_id: member.user_id
+            }
+          });
+          if (!anonInfo && oldChannelId && member.user_id !== req.userId) {
+            let oldAnonInfo = await ChatAnonUserInfo.findOne({
+              where: {
+                channel_id: oldChannelId,
+                my_anon_user_id: {
+                  [Sequelize.Op.ne]: req.userId
+                }
+              }
+            });
+            if (oldAnonInfo) {
+              anonInfo = oldAnonInfo;
+              await ChatAnonUserInfoFunction.createChatAnonUserInfo(
+                ChatAnonUserInfo,
+                findOrCreateChannel.channel.id,
+                member.user_id,
+                req?.userId,
+                {
+                  anon_user_info_color_code: anonInfo.anon_user_info_color_code,
+                  anon_user_info_color_name: anonInfo.anon_user_info_color_name,
+                  anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
+                  anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name
+                }
+              );
+            }
+          }
+  
+          let anonUserInfo = {};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->